pkgs/build-support/deno/fetch-deno-deps/default.nix

```diff
@@ -0,0 +1,257 @@
+    excludeDrvArgNames = [
+      "denoLock"
+    ];
+    extendDrvArgs =
+      finalAttrs:
+      {
+        denoLock,
+        pname,
+        version,
+        ...
+      }@args:
+      {
+        name = args.name or "denoDeps-transformedDenoLock-${finalAttrs.pname}-${finalAttrs.version}";
+        src = args.src or null;
+        dontUnpack = args.dontUnpack or true;
+        buildPhase =
+          args.buildPhase or ''
+            mkdir -p $out
+            lockfile-transformer \
+              --deno-lock-path ${denoLock} \
+              --common-lock-jsr-path $out/jsr.json \
+              --common-lock-npm-path $out/npm.json \
+              --common-lock-https-path $out/https.json;
+          '';
+        nativeBuildInputs =
+          args.nativeBuildInputs or [
+            fetch-deno-deps-scripts
+          ];
+
+        meta =
+          args.meta or {
+            maintainers = [ lib.maintainers.aMOPel ];
+          };
+      };
+  };
+
+  singleFodFetcher = lib.extendMkDerivation {
+    constructDrv = stdenvNoCC.mkDerivation;
+    # don't let these bleed into the derivation environment
+    excludeDrvArgNames = [
+      "transformedDenoLock"
+      "denoLock"
+      "hash"
+      "fodNameGenerator"
+      "enableAutoFodInvalidation"
+      "argNamesForFodInvalidation"
+    ];
+    extendDrvArgs =
+      finalAttrs:
+      {
+        transformedDenoLock,
+        denoLock,
+        hash,
+        fodNameGenerator,
+        enableAutoFodInvalidation,
+        argNamesForFodInvalidation,
+        pname,
+        version,
+        ...
+      }@args:
+      let
+        argsForFodInvalidation = lib.filterAttrs (
+          name: _: builtins.any (otherName: name == otherName) argNamesForFodInvalidation
+        ) finalAttrs;
+        defaultName = "denoDeps-fetched-${finalAttrs.pname}-${finalAttrs.version}";
+      in
+      {
+        name =
+          if enableAutoFodInvalidation then
+            fodNameGenerator argsForFodInvalidation finalAttrs.pname finalAttrs.version
+          else
+            args.name or defaultName;
+        src = args.src or null;
+        dontUnpack = args.dontUnpack or true;
+        buildPhase =
+          args.buildPhase or ''
+            single-fod-fetcher \
+              --common-lock-jsr-path ${transformedDenoLock}/jsr.json \
+              --common-lock-npm-path ${transformedDenoLock}/npm.json \
+              --common-lock-https-path ${transformedDenoLock}/https.json \
+              --out-path-prefix $out;
+            cp ${denoLock} $out/deno.lock
+          '';
+        nativeBuildInputs =
+          args.nativeBuildInputs or [
+            fetch-deno-deps-scripts
+          ];
+        SSL_CERT_FILE = args.SSL_CERT_FILE or "${cacert}/etc/ssl/certs/ca-bundle.crt";
+
+        outputHashMode = args.outputHashMode or "recursive";
+        outputHash = args.outputHash or hash;
+        outputHashAlgo = args.outputHashAlgo or "sha256";
```

Comment:
Please make this: 

1. Uncondtionally use `hash`
2. Unconditoinally pass `""` to `outputHashAlgo`

This forces users to use SRI hashes which is what Nixpkgs wants.